extensions/amp-animation/0.1/web-animations.js

```diff
@@ -315,35 +333,123 @@ class Scanner {
 
 
 /**
+ * Builds animation runners based on the provided spec.
+ */
+export class Builder {
+  /**
+   * @param {!Window} win
+   * @param {!Node} rootNode
+   * @param {string} baseUrl
+   * @param {!../../../src/service/vsync-impl.Vsync} vsync
+   * @param {!../../../src/service/resources-impl.Resources} resources
+   */
+  constructor(win, rootNode, baseUrl, vsync, resources) {
+    /** @const @private */
+    this.css_ = new CssContextImpl(win, rootNode, baseUrl);
+
+    /** @const @private */
+    this.vsync_ = vsync;
```

Comment:
Easier to test. This is an internal class - the main reason is to split the code and test.